### PR TITLE
std.range: document public methods

### PR DESCRIPTION
**Pull request by wilzbach**
follow-up to #4312

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -2887,13 +2905,37 @@ private:
     UT _value;
 
 public:
-    @property inout(T) front() inout { return _value; }
-    @property inout(T) back() inout { return _value; }
+    ///
```
Comment (JackStouffer) [inline std/range/package.d]: Should have comment like "Range primitives"

File: std/range/package.d [hunk 2]
```diff
@@ -2887,13 +2905,37 @@ private:
...
+    @property inout(T) front() inout
+    {
+        return _value;
+    }
+
+    ///
```
Comment (JackStouffer) [inline std/range/package.d]: All other range primitives should ditto the above

Comment (wilzbach) [inline std/range/package.d]: Is this the common way to document these?
I thought it's preferred to just explicitly document these methods and their meaning is intrinsic.

Comment (JackStouffer) [inline std/range/package.d]: I see a mix of both in the docs, I'm just not a fan of docstrings with no words

Comment (wilzbach) [inline std/range/package.d]: > I see a mix of both in the docs, I'm just not a fan of docstrings with no words

Let's wait for feedback from other people here - I am more on the side of supplying unnecessary information bloats and is less helpful.

Comment (wilzbach) [inline std/range/package.d]: > Let's wait for feedback from other people here - I am more on the side of supplying unnecessary information bloats and is less helpful.

Does anyone have an opinion on whether range primitives need a comment or just documenting it with `///` is better?

Comment (schveiguy) [inline std/range/package.d]: If you are going to document, please put something.

I think it's fine to do "Range primitives" on the first one, and then ditto the rest. As of now, the empty docs take up a LOT of space: http://dtest.thecybershadow.net/artifact/website-fc52239a89c0f31ce8fab3844f156b775682a9e6-929fb93b8eac814d30aee0920e8c35b9/web/phobos-prerelease/std_range.html#.Repeat

File: std/range/package.d [hunk 3]
```diff
@@ -2887,13 +2905,37 @@ private:
...
     enum bool empty = false;
+
+    ///
     void popFront() {}
+
+    ///
     void popBack() {}
+
+    ///
     @property auto save() inout { return this; }
```
Comment (schveiguy) [inline std/range/package.d]: Why did you expand all the other primitives, but not this one? And why expand them at all? Is there a rule on this in the style guidelines? I prefer the original way better.

**Comment by wilzbach** [comment]
> Why did you expand all the other primitives, but not this one? And why expand them at all? Is there a rule on this in the style guidelines?

I think most of the code had the "long style", but I don't remember anymore.
Reverted the changes as this should only be about documenting all public methods + addressed other comments.

File: std/range/package.d [hunk 4]
```diff
@@ -1936,23 +1953,13 @@ if (isInputRange!(Unqual!Range) &&
         }
     }
 
-    // Nonstandard
+    /// Nonstandard
```
Comment (JackStouffer) [inline std/range/package.d]: You're going to want something a little more descriptive here. What about this is non-standard and why does that matter?

**Comment by JackStouffer** [comment]
Everything else looks good

**Comment by JackStouffer** [comment]
LGTM

File: std/range/package.d [hunk 5]
```diff
@@ -1964,6 +1975,17 @@ if (isInputRange!(Unqual!R) && !isInfinite!(Unqual!R) && hasSlicing!(Unqual!R) &
     return input[0 .. min(n, input.length)];
 }
 
+/**
+This template simply aliases itself to R and is useful for consistency in
```
Comment (andralex) [inline std/range/package.d]: R in code font

**Comment by andralex** [comment]
will pull, feel free to address the nit in a later pass

**Pull request merged**